src/common/tests/lru.c

```diff
@@ -118,15 +118,31 @@ main(int argc, char **argv)
 		exit(-1);
 	}
 
-	csize	 = (int)strtol(argv[1], (char **)NULL, 10);
-	num_keys = (int)strtol(argv[2], (char **)NULL, 10);
+	csize = strtol(argv[1], (char **)NULL, 10);
+	num_keys = strtol(argv[2], (char **)NULL, 10);
+
+	if (csize < 0 || csize > INT_MAX) {
+		rc = -DER_INVAL;
+		D_ERROR("Invalid cell size\n");
+		D_GOTO(exit, rc);
+	}
 
 	rc = daos_lru_cache_create(csize, D_HASH_FT_RWLOCK,
 				   &uint_ref_llink_ops,
 				   &tcache);
 	if (rc)
 		D_ASSERTF(0, "Error in creating lru cache\n");
 
+	/* make sure csize and num_keys can
+	 * fit into int variable, since
+	 * they used to be cast to int values
+	 */
```

Comment:
Minor, if you need to push again: This comment refers to csize and num_key, but the csize check is above. Maybe move the comment there?